paddle/fluid/operators/uniform_random_inplace_op.cc

```diff
@@ -0,0 +1,128 @@
+    PADDLE_ENFORCE_LT(
+        ctx->Attrs().Get<float>("min"), ctx->Attrs().Get<float>("max"),
+        platform::errors::InvalidArgument(
+            "The uniform_random's min must less then max. But received min = "
+            "%f great than or equal max = %f.",
+            ctx->Attrs().Get<float>("min"), ctx->Attrs().Get<float>("max")));
+    PADDLE_ENFORCE_GE(ctx->Attrs().Get<int>("diag_num"), 0,
+                      platform::errors::InvalidArgument(
+                          "The uniform_random's diag_num must greater than or "
+                          "equal 0. But recevied diag_num (%d) < 0.",
+                          ctx->Attrs().Get<int>("diag_num")));
+    PADDLE_ENFORCE_GE(ctx->Attrs().Get<int>("diag_step"), 0,
+                      platform::errors::InvalidArgument(
+                          "The uniform_random's diag_step must greater than or "
+                          "equal 0. But recevied diag_step (%d) < 0.",
+                          ctx->Attrs().Get<int>("diag_step")));
+    auto xdim = ctx->GetInputDim("X");
+    ctx->SetOutputDim("Out", xdim);
+  }
+
+ protected:
+  framework::OpKernelType GetExpectedKernelType(
+      const framework::ExecutionContext &ctx) const override {
+    return framework::OpKernelType(
+        OperatorWithKernel::IndicateVarDataType(ctx, "X"), ctx.GetPlace());
+  }
+};
+
+template <typename T>
+class CPUUniformRandomInplaceKernel : public framework::OpKernel<T> {
+ public:
+  void Compute(const framework::ExecutionContext &ctx) const override {
+    auto out_var = ctx.OutputVar("Out");
+    auto *tensor = out_var->GetMutable<framework::LoDTensor>();
+    T *data = tensor->mutable_data<T>(ctx.GetPlace());
+    int64_t size = tensor->numel();
+    std::uniform_real_distribution<T> dist(
+        static_cast<T>(ctx.Attr<float>("min")),
+        static_cast<T>(ctx.Attr<float>("max")));
+    auto engine = paddle::framework::GetCPURandomEngine(
+        static_cast<unsigned int>(ctx.Attr<int>("seed")));
+    for (int64_t i = 0; i < size; ++i) {
+      data[i] = dist(*engine);
+    }
+  }
+};
+
+class UniformRandomInplaceOpVarTypeInference
+    : public framework::VarTypeInference {
+ public:
+  void operator()(framework::InferVarTypeContext *ctx) const override {}
+};
+
+}  // namespace operators
+}  // namespace paddle
+
+DECLARE_INPLACE_OP_INFERER(UniformRandomInplaceInToOut, {"X", "Out"});
+REGISTER_OPERATOR(
+    uniform_random_inplace, paddle::operators::UniformRandomInplaceOp,
+    paddle::operators::UniformRandomInplaceOpMaker,
+    paddle::framework::EmptyGradOpMaker<paddle::framework::OpDesc>,
+    paddle::framework::EmptyGradOpMaker<paddle::imperative::OpBase>,
+    paddle::operators::UniformRandomInplaceOpVarTypeInference,
+    UniformRandomInplaceInToOut);
+REGISTER_OP_CPU_KERNEL(
+    uniform_random_inplace,
+    paddle::operators::CPUUniformRandomInplaceKernel<float>,
+    paddle::operators::CPUUniformRandomInplaceKernel<double>);
```

Comment:
Please refer to other inplace operator's implementation, under normal situation, it can be shared with files of non-inplace, so no need to add uniform_random_inplace_op.cc, uniform_random_inplace_op.cu and uniform_random_inplace_op_xpu.cc

Reply:
Because the parameters between paddle.uniform and paddle.Tensor.uniform_ are different, non-inplace version Uniform Kernel can not be directly reused.